shell/app/node_main.cc

```diff
@@ -221,6 +171,29 @@ int NodeMain(int argc, char* argv[]) {
       return result->exit_code();
     }
 
+#if BUILDFLAG(IS_LINUX)
+    // On Posix, initialize crashpad after Nodejs init phase so that
+    // crash and termination signal handlers can be set by the crashpad client.
+    if (!pid_string.empty()) {
+      auto* command_line = base::CommandLine::ForCurrentProcess();
+      command_line->AppendSwitchASCII(
+          crash_reporter::switches::kCrashpadHandlerPid, pid_string);
```

Comment:
just want to double check that this doesn't actually affect `process.argv` in the child process? (should we have a test for that...?)

Reply:
I wonder if the right approach here is to check to see if the PID switch exists/has an existing value, and only append the switch if it doesn't? Something like this?:
```suggestion
      if (!command_line->HasSwitch(crash_reporter::switches::kCrashpadHandlerPid)) {
        command_line->AppendSwitchASCII(
            crash_reporter::switches::kCrashpadHandlerPid, pid_string);
      }
```

Reply:
Agree with adding a test, I am pretty sure this will not affect `process.argv` since we are setting the flag after Nodejs has processed the args.

@VerteDinde that check is not needed since there is no other candidate that can add the flag before `NodeMain`, `NodeMain` should be responsible for adding that flag or overwrite any if present.

Reply:
Just tested this manually and it looks like Deepak's right 🙂 I'll add a test

Reply:
Test added in [b449baf](https://github.com/electron/electron/pull/36460/commits/b449baf781d9da8e7c0cdf4a38a27fd184670543)